### PR TITLE
Fix path to watchSources in watch.coffee

### LINKED ISSUE #2928: watch.coffee has bad path for bokehjs watchSources (gulp watch is broken)
Breaking widgets out of bokehjs changed the nested structure of the module.exports object in paths.coffee, so the watchSources path in watch.coffee is wrong.

The current watch path is:
```coffeescript
...
gulp.task "watch", ->
  gulp.watch "#{paths.coffee.watchSources}", ->
    runSequence("scripts:build")
  ...
```

It should be:

```coffeescript
...
gulp.talk "watch", ->
  gulp.watch "#{paths.bokehjs.coffee.watchSources}", ->
    runSequence("scripts:build")
```

### PR DESCRIPTION
**Pull request by canavandl**
issues: fixes #2928 

Corrects path to watchSources in watch.coffee

### CODE REVIEW

**Comment by damianavila** [comment]
LGTM, pinging @mattpap for another view before merging...

File: bokehjs/gulp/tasks/watch.coffee [hunk 1]
```diff
@@ -5,7 +5,7 @@ runSequence = require "run-sequence"
 paths = require "../paths"
 
 gulp.task "watch", ->
-  gulp.watch "#{paths.coffee.watchSources}", ->
+  gulp.watch "#{paths.bokehjs.coffee.watchSources}", ->
     runSequence("scripts:build")
   gulp.watch "#{paths.css.watchSources}", ->
```
Comment (mattpap) [inline bokehjs/gulp/tasks/watch.coffee]: This needs to be fixed as well. Should be `less`.

**Comment by mattpap** [comment]
I think that `{bokehjs,widgets}.coffee.watchSources` should be moved to `coffee.watchSources` and then `bokehjs` part removed from the appropriate task. Otherwise, `widgets.coffee.watchSources` is not going to be ever used.

**Comment by mattpap** [comment]
As we attempt to fix things, this whole `watch` task's implementation is suspicious to me, because you have `runSequence()` twice, i.e. two streams, and gulp allows only one. Otherwise, you have to use stream merging or callbacks. I doubt this code does what it was intended to do. Perhaps no one noticed, because this task is rarely used and it effectively watches only coffeescript sources anyway (see my earlier comment).

**Comment by canavandl** [comment]
Three things:

1) Per @mattpap, I refactored the paths.coffee object to the following. I think it's a lot better.

```coffeescript
  coffee:
    bokehjs:
      destination:
        ...
    widgets:
      destination:
        ...
    sources: [
        "./src/coffee/main.coffee"
        "./src/coffee/widget/main.coffee"
    ]
    watchSources: [
      "./src/coffee/**/**"
    ]
```

2) I changed the gulp `watch` task to use the less files as the watchSources (not CSS), so that should work now too.

3) About the overall `watch` task implementation, it appears (to me) that runSequence is sent as a callback to `gulp.watch` so it's able to build CSS/JS concurrently. It "works" for me, but I'm no gulp expert.

**Comment by bryevdv** [comment]
@brendancol @mistakevin @mattpap thoughts? I don't have context/opinions here.

**Comment by mattpap** [comment]
I will have to make sure what this `runSequence()` actually does, because it doesn't fit into any of patterns that I know gulp tasks support. Anyway, we can merge this PR as-is and investigate this later.

**Pull request merged**